src/main/java/uk/gov/hmcts/reform/sandl/snlrules/services/DroolsService.java

```diff
@@ -30,11 +36,21 @@ public void init() {
         KieContainer kieContainer = kieServices.getKieClasspathContainer();
         rulesSession = kieContainer.newKieSession(droolsConfiguration.getRulesKSession());
 
-        rulesSession.addEventListener(new FactsChangedEventListener());
-        rulesSession.addEventListener(new RulesChangedEventListener());
+        factsChangedEventListener = new FactsChangedEventListener();
+
+        rulesSession.addEventListener(factsChangedEventListener);
```

Comment:
Could you move instantiation of FactsChangedEventListener to addEventListener and also remove private field?

Reply:
no :) it's used in methods below